dev-lang/swi-prolog/swi-prolog-9.1.2.ebuild

```diff
@@ -62,72 +82,82 @@ src_prepare() {
 		eapply "${WORKDIR}"/${PV}
 	fi
 
-	sed -e "s|\(SWIPL_INSTALL_PREFIX\)   lib/.*)|\1   $(get_libdir)/swipl)|" \
+	sed \
+		-e "s|\(SWIPL_INSTALL_PREFIX\)   lib/.*)|\1   $(get_libdir)/swipl)|" \
 		-e "s|\(SWIPL_INSTALL_CMAKE_CONFIG_DIR\) lib/|\1   $(get_libdir)/|" \
 		-i CMakeLists.txt || die
 
+	sed "s/ -Werror//g" \
+		-i cmake/GCCBuiltins.cmake \
+		-i cmake/Config.cmake \
+		-i packages/ssl/CMakeLists.txt || die
+
 	local ncurses_lib_flags=$($(tc-getPKG_CONFIG) --libs ncurses)
 	sed -i "/project(SWI-Prolog)/a set(CMAKE_REQUIRED_LIBRARIES \${CMAKE_REQUIRED_LIBRARIES} ${ncurses_lib_flags})" CMakeLists.txt || die
 	sed -i "s:\${CURSES_LIBRARIES}:${ncurses_lib_flags}:" src/CMakeLists.txt || die
 
+	java-pkg-opt-2_src_prepare
 	cmake_src_prepare
 }
 
 src_configure() {
 	append-flags -fno-strict-aliasing
 	use debug && append-flags -DO_DEBUG
 
+	if use pgo; then
+		CMAKE_BUILD_TYPE="PGO"
+		append-flags -Wno-error=coverage-mismatch
+	fi
+
 	local mycmakeargs=(
+		-DBUILD_TESTING=$(usex test)
+		-DINSTALL_DOCUMENTATION=$(usex doc)
 		-DSWIPL_INSTALL_PREFIX=$(get_libdir)/swipl
-		-DUSE_GMP=$(usex gmp)
-		-DINSTALL_DOCUMENTATION=$(use doc && usex archive)
-		-DSWIPL_PACKAGES_BASIC=$(usex !minimal)
 		-DSWIPL_PACKAGES_ARCHIVE=$(usex archive)
-		-DSWIPL_PACKAGES_ODBC=$(usex odbc)
+		-DSWIPL_PACKAGES_BASIC=$(usex !minimal)
 		-DSWIPL_PACKAGES_BDB=$(usex berkdb)
+		-DSWIPL_PACKAGES_JAVA=$(usex java)
+		-DSWIPL_PACKAGES_ODBC=$(usex odbc)
 		-DSWIPL_PACKAGES_PCRE=$(usex pcre)
-		-DSWIPL_PACKAGES_YAML=$(usex yaml)
+		-DSWIPL_PACKAGES_QT=$(usex gui)
 		-DSWIPL_PACKAGES_SSL=$(usex ssl)
-		-DSWIPL_PACKAGES_JAVA=$(usex java)
-		-DSWIPL_PACKAGES_QT=$(usex qt5)
+		-DSWIPL_PACKAGES_TERM=$(usex cli)
+		-DSWIPL_PACKAGES_TIPC=$(usex ipc)
 		-DSWIPL_PACKAGES_X=$(usex X)
-		-DSWIPL_PACKAGES_TERM=$(if use libedit || use readline; then echo yes; else echo no; fi)
+		-DSWIPL_PACKAGES_YAML=$(usex yaml)
+		-DUSE_GMP=$(usex gmp)
+		-DUSE_TCMALLOC=OFF
+		$(cmake_use_find_package qt6 Qt6)
```

Comment:
This looks to be related to the `-DSWIPL_PACKAGES_QT=$(usex gui)` part. I'll order them around, thanks!